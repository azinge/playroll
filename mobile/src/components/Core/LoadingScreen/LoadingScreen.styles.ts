import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    backgroundColor: 'purple',
    flex: 2,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
