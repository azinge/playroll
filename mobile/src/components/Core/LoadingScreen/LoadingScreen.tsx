/**
 * LoadingScreen
 */

import * as React from 'react';
import { View, Dimensions, ImageBackground } from 'react-native';
import {
  NavigationScreenProp,
  StackActions,
  NavigationActions,
} from 'react-navigation';

import styles from './LoadingScreen.styles';
import { GetCurrentUserQuery } from '../../../graphql/requests/User';
import NavigationService from '../../../services/NavigationService';
import NotificationService from '../../../services/NotificationService';

export interface Props {
  navigation?: NavigationScreenProp<{}>;
}

interface State {
  deviceToken: string;
}
export default class LoadingScreen extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      deviceToken: undefined,
    };
  }
  async componentDidMount() {
    const deviceToken = await NotificationService.registerForPushNotificationsAsync();
    this.setState({ deviceToken });
  }
  render() {
    const { width, height } = Dimensions.get('window');
    return (
      <View style={styles.container}>
        <GetCurrentUserQuery
          skip={this.state.deviceToken === undefined}
          variables={{
            deviceToken: this.state.deviceToken
              ? this.state.deviceToken
              : undefined,
          }}
        >
          {({ loading, error }) => {
            if (!loading && this.state.deviceToken !== undefined) {
              setTimeout(() => {
                NavigationService.dispatch(
                  StackActions.reset({
                    key: null,
                    index: 0,
                    actions: [
                      NavigationActions.navigate({
                        routeName: error ? 'Auth' : 'Main',
                      }),
                    ],
                  })
                );
              }, 1250);
            }
            return null;
          }}
        </GetCurrentUserQuery>
        <ImageBackground
          source={require('../../../assets/loading.png')}
          resizeMode='cover'
          style={{ height: height, width: width }}
        />
      </View>
    );
  }
}
