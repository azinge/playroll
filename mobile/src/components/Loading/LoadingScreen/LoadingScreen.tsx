/**
 * LoadingScreen
 */

import * as React from "react";
import {
  Text,
  View,
  Image,
  SafeAreaView,
  Dimensions,
  ImageBackground,
} from "react-native";
import {
  NavigationScreenProp,
  StackActions,
  NavigationActions,
} from "react-navigation";

import styles from "./LoadingScreen.styles";

export interface Props {
  navigation?: NavigationScreenProp<{}>;
}

interface State {}
export default class LoadingScreen extends React.Component<Props, State> {
  componentDidMount() {
    setTimeout(() => {
      this.props.navigation &&
        this.props.navigation.dispatch(
          StackActions.reset({
            key: null,
            index: 0,
            actions: [NavigationActions.navigate({ routeName: "Auth" })],
          })
        );
    }, 1250);
  }
  render() {
    const { width, height } = Dimensions.get("window");
    return (
      <View style={styles.container}>
        <ImageBackground
          source={require("../../../assets/loading.png")}
          resizeMode="cover"
          style={{ height: height, width: width }}
        />
      </View>
    );
  }
}
