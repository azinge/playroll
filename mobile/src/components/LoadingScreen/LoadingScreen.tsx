/**
 * LoadingScreen
 */

import * as React from "react";
import {
  Text,
  View,
  Image,
  SafeAreaView,
  Dimensions,
  ImageBackground,
} from "react-native";

export default class LoadingScreen extends React.Component {
  render() {
    const { width, height } = Dimensions.get("window");

    return (
      <View
        style={{
          backgroundColor: "purple",
          flex: 2,
          justifyContent: "center",
          alignItems: "center",
        }}
      >
        <ImageBackground
          source={require("../../assets/loading.png")}
          resizeMode="cover"
          style={{ height: height, width: width }}
        />
      </View>
    );
  }
}
