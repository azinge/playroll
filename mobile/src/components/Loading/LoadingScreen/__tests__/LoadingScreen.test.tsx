import * as React from "react";
import renderer from "react-test-renderer";

import LoadingScreen from "../LoadingScreen";

it("renders correctly", () => {
  const tree = renderer.create(<LoadingScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
