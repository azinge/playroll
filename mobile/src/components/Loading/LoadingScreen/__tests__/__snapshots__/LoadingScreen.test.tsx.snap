// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "purple",
      "flex": 2,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "height": 1334,
        "width": 750,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "testUri": "../../../src/assets/loading.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
